src/lib/src/common/workspace.spec.ts

```diff
@@ -1,331 +1,442 @@
+      stopOn: (currentState) => currentState !== stateBefore
+    }
+    const workspace = createWorkspaceWithRootFolder(observer.path);
+    const invocationsUntilChange = 3;
+    let invocations = 0;
+
+    // when
+    workspace.observeMarker(observer);
+    tick();
+
+    // then
+    expect(workspace.getMarkerValue(observer.path, observer.field)).toEqual(stateAfter);
+    expect(invocations).toEqual(invocationsUntilChange + 1);
+  }));
+
+  it('stops polling when the path disappears from the workspace', fakeAsync(() => {
+    // given
+    const stateBefore = ElementState.Running;
+    const stateAfter = ElementState.LastRunFailed;
+    const observer: MarkerObserver<ElementState> = {
+      path: 'sample/path',
+      field: 'testStatus',
+      observe: () => new Promise((resolve) => setTimeout(resolve, 1))
+        .then(() => Promise.resolve(invocations++ < invocationsUntilChange ? stateBefore : stateAfter)),
+      stopOn: (currentState) => currentState !== stateBefore
+    }
+    const workspace = createWorkspaceWithRootFolder(observer.path);
+    const invocationsUntilChange = 3;
+    const invocationsUntilReload = 2;
+    let invocations = 0;
+
+    workspace.observeMarker(observer);
+    tick(invocationsUntilReload);
+
+    // when
+    workspace.reload({name: '', path: 'unobserved/path', type: ElementType.File, children: []});
+    tick();
+
+    // then
+    expect(invocations).toEqual(invocationsUntilReload);
+    expect(workspace.getMarkerValue(observer.path, observer.field)).toEqual(stateBefore);
+    flush();
+  }));
+
+  it('throws an error when the path does not exist', () => {
+    // given
+    const observer: MarkerObserver<ElementState> = {
+      path: 'non/existing/path',
+      field: 'testStatus',
+      observe: () => Promise.resolve(ElementState.LastRunSuccessful),
+      stopOn: () => true
+    }
+    const workspace = createWorkspaceWithRootFolder('existing/path');
+
+    // when + then
+    expect(() => workspace.observeMarker(observer))
+    .toThrowError(`There is no element with path "${observer.path}" in this workspace.`);
+  });
+
+  it('continues polling when the observable produces errors', fakeAsync(() => {
+    // given
+    const state = Array(4).fill(ElementState.Running).concat([ElementState.LastRunSuccessful]);
+    const observer: MarkerObserver<ElementState> = {
+      path: 'sample/path',
+      field: 'testStatus',
+      observe: () => invocations++ % 2 ? Promise.resolve(state[invocations]) : Promise.reject('failure'),
+      stopOn: (currentState) => currentState !== ElementState.Running
+    }
+    const workspace = createWorkspaceWithRootFolder(observer.path);
+    let invocations = 0;
+    workspace.setMarkerValue(observer.path, observer.field, state[0]);
+
+    // when
+    workspace.observeMarker(observer)
+    tick();
+
+    // then
+    expect(invocations).toEqual(4);
+    expect(workspace.getMarkerValue(observer.path, observer.field)).toEqual(state[invocations]);

```

Comment:
toEqual(ElementState.LastRunSuccessful) which is semantically the same but makes it more explicit what is expected

Reply:
my reasoning here was that the specific value does not really matter, but rather that it corresponds with the expected state after `invocations` number of invocations.